### ISSUE TITLE
[Question] Can this be used to save a dynamic assembly to disk? 

### ISSUE THREAD

**Issue by d-dizhevsky** [0xd4d/dnlib#209]
As it is currently not possible to use ref emit `AssemblyBuilder.Save` from .NET Standard 2.0 - can this lib be used for that? 
I'm going to do code gen using ref emit, will I then be able to save the resulting assembly to disk using this?

**Comment by 0xd4d** [comment]
Anything is possible if you have enough time. dnSpy's debugger for instance can do that. The easiest solution is to of course debug it on .NET Framework so reflection can save the dynamic assembly to disk.

**Comment by 0xd4d** [comment]
Also are you sure .NET Core doesn't support this functionality? If it does, you could use reflection to access the right method and save it.

**Comment by d-dizhevsky** [comment]
Unfortunately .NET Standard doesn't support this at the moment (and afaik these methods are even not in the .net standard vNext roadmap). see https://github.com/dotnet/corefx/issues/4491

What do you mean by `debug it on .NET Framework`?

**Comment by 0xd4d** [comment]
You could write a simple .NET Framework app that uses your netstandard library and use reflection to access the Save() method. I assume you only need it for debugging purposes.

**Comment by per-samuelsson** [comment]
FWIW, for anyone landing here, I couldn't get it to work saving a generating assembly to disk. Built `dnlib` and made a simple prototype, something like:

```cs
var assemblyName = new AssemblyName("TestAssembly");
var builder = AssemblyBuilder
    .DefineDynamicAssembly(assemblyName, AssemblyBuilderAccess.Run)
    .DefineDynamicModule(assemblyName.FullName);

builder.DefineType("Test").CreateType();

var mod = ModuleDefMD.Load(builder.Assembly.ManifestModule);
mod.Write(@"C:\saved-assembly.dll");
```

Loading module failed with

```
An unhandled exception of type 'System.InvalidOperationException' occurred in dnlib.dll: 'Module RefEmit_InMemoryManifestModule has no HINSTANCE'
```

Ran on .NET Core 3.0 RC1 (Win).

**Comment by 0xd4d** [comment]
It's not supported, you have to do it yourself, it's what dnSpy does. There's a nuget package (I've not tried it myself), see https://github.com/dotnet/corefx/issues/4491#issuecomment-469009414

**Comment by per-samuelsson** [comment]
> It's not supported, you have to do it yourself, it's what dnSpy does.

I see, thanks for clarifying.

>  There's a nuget package (I've not tried it myself), see dotnet/corefx#4491 (comment)

Tried that before I tried this, and failed there too unfortunately. Failed on this:

https://github.com/Lokad/ILPack/issues/107#issuecomment-533771081

and that was reported some time ago, and don't seem to be fixed.

So seems this story (saving generated assemblies to disk on net core) is kind of status quo since a few years back.
